core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -163,6 +164,7 @@ private[spark] class CoarseGrainedExecutorBackend(
 
 private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
+  var executorBackend : CoarseGrainedExecutorBackend = _
```

Comment:
Looks like an ugly workaround. Can't you override the thread's uncaught handler?

Alternatively adding a "getUncaughtHandler" to ExecutorBackend will be a better idea.